kroxylicious/src/main/java/io/kroxylicious/proxy/internal/DefaultFilterContext.java

```diff
@@ -223,6 +223,7 @@ private void forwardShortCircuitResponse(ResponseHeaderData header, ApiMessage r
                     "Attempt to respond with ApiMessage of type " + ApiKeys.forId(response.apiKey()) + " but request is of type " + decodedFrame.apiKey());
         }
         DecodedResponseFrame<?> responseFrame = new DecodedResponseFrame<>(decodedFrame.apiVersion(), decodedFrame.correlationId(), header, response);
+        decodedFrame.transferBuffers(responseFrame);
```

Comment:
I think you mentioned elsewhere why its this way round but it feels odd that we are transferring a responseFrame *to* a request frame. Naively I would expect us to keep the request frame around until we had finished with the response not the other way around.

Reply:
It's the other way around, I've renamed the method to `decodedFrame.transferBuffersTo(responseFrame)`

Reply:
I guess part of what confused me so the pushing of buffers rather than pulling but I guess it has to be that way round for access